File: apps/pass-at-k-study/benchmark/tasks/legate/l2_dependent_pipeline.py
# Task: l2_dependent_pipeline — matches L2DependentPipeline TEST_CASES.
import math
import sys

_s = 7


def randf01() -> float:
    global _s
    _s = (_s * 1103515245 + 12345) & 0x7FFFFFFF
    return float(_s & 0xFFFFFF) / float(0x1000000)


def pipeline(n: int) -> float:
    nn = n * n
    a = [randf01() for _ in range(nn)]
    mx = max(a) if a else 0.0
    if mx < 1e-6:
        mx = 1e-6
    return sum(x / mx for x in a)


def main() -> int:
    global _s
    _s = 7
    for n in (256, 1024, 4096):
        v = pipeline(n)
        if not math.isfinite(v):
            print("FAIL")
            return 1
    print("PASS")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: apps/pass-at-k-study/benchmark/tasks/iris/l2_dependent_pipeline.cpp
// Task: l2_dependent_pipeline — matches L2DependentPipeline TEST_CASES.
#include <cmath>
#include <cstdint>
#include <iostream>
#include <vector>

static std::uint32_t g_seed = 7u;
static float randf01() {
  g_seed = g_seed * 1103515245u + 12345u;
  return float(g_seed & 0xffffffu) / float(0x1000000u);
}

static float pipeline(int N) {
  size_t n = size_t(N) * size_t(N);
  std::vector<float> A(n);
  for (size_t i = 0; i < n; ++i) A[i] = randf01();
  float mx = 0.f;
  for (float x : A)
    if (x > mx) mx = x;
  if (mx < 1e-6f) mx = 1e-6f;
  float s = 0.f;
  for (float x : A) s += x / mx;
  return s;
}

int main() {
  int Ns[] = {256, 1024, 4096};
  g_seed = 7u;
  for (int N : Ns) {
    float v = pipeline(N);
    if (!std::isfinite(v)) {
      std::cout << "FAIL\n";
      return 1;
    }
  }
  std::cout << "PASS\n";
  return 0;
}
